hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -110,6 +110,161 @@ public static function findById($id) {
   }
 
   /**
+   * Get 'length_of_service' in days for given Contact ID, and optionally
+   * Date and Break (allowed number of days between Contracts).
+   * 
+   * @param int $contactId
+   * @param string $date
+   * @param int $break
+   * @throws Exception
+   * @return int
+   */
+  public static function getLengthOfService($contactId, $date = null, $break = 14) {
```

Comment:
This method is very long, you should split it into smaller private methods

*(kind of pseudo-code)*
```php
function getContractDates($contracts) {
    foreach ($contracts['values'] as $contract) {
        // ...
    }

    ksort($dates);
    return $dates;
}

function getServiceStartAndDateDates($allContractDates) {
    foreach ($allContractDates as $startDate => $endDate) {
        // ...
    }

    return array('start_date' => $serviceStartDate, 'end_date' => $serviceEndDate);
}

function calculateLength($startAndEnd) {
    // ...
    return $dDiff->days;
}
```